packages/mobile/src/screens/edit-collection-screen/AdvancedAlbumScreen.tsx

```diff
@@ -0,0 +1,67 @@
+import { advancedAlbumMessages as messages } from '@audius/common/messages'
+import { useField } from 'formik'
+
+import {
+  Divider,
+  Flex,
+  IconCalendarMonth,
+  IconIndent,
+  Text
+} from '@audius/harmony-native'
+import { DateTimeInput } from 'app/components/core'
+import { HarmonyTextField } from 'app/components/fields'
+import { FormScreen } from 'app/screens/form-screen'
+
+export const AdvancedAlbumScreen = () => {
+  const [{ value: upc }, { touched }] = useField('upc')
+  const [{ value: isHidden }] = useField('is_private')
+  const [{ value: releaseDate, onChange }] = useField('release_date')
+
+  const error = !upc || /^\d{12}$/.test(upc) ? null : messages.upcInputError
```

Comment:
nbd, I'd probably wrap this and the below into a UPCField component which handles live validation and separates the concern from future Advanced field siblings, leaving the form layer on this component.

Something like:
```
<FormScreen>
	<UPCField />
	{ !isHidden && <ReleaseDateField /> }
</FormScreen>
```

Reply:
you'd have to disable submit via validation instead of `disableSubmit` though

all super optional imo